src/qibo/base/gates.py

```diff
@@ -2,13 +2,17 @@
 # @authors: S. Carrazza and A. Garcia
 from qibo import config
 from qibo.config import raise_error
-from typing import Dict, List, Optional, Sequence, Tuple
+from typing import Dict, List, Iterable, Optional, Sequence, Tuple
 
 QASM_GATES = {"h": "H", "x": "X", "y": "Y", "z": "Z",
               "rx": "RX", "ry": "RY", "rz": "RZ",
-              "cx": "CNOT", "swap": "SWAP",
-              "crz": "CZPow", "ccx": "TOFFOLI"}
-PARAMETRIZED_GATES = {"rx", "ry", "rz", "crz"}
+              "u1": "U1", "u2": "U2", "u3": "U3",
+              "cx": "CNOT", "swap": "SWAP", "cz": "CZ",
+              "crx": "CRX", "cry": "CRY", "crz": "CRZ",
+              "cu1": "CU1", "cu3": "CU3",
+              "ccx": "TOFFOLI"}
+PARAMETRIZED_GATES = {"rx", "ry", "rz", "u1", "u2", "u3",
+                      "crx", "cry", "crz", "cu1", "cu3"}
```

Comment:
I think cu2 is missing here and above, right?

Reply:
That's correct but it is actually on purpose. The reason is that this dictionary and set are only used by the `to_qasm`/`from_qasm` methods and I think that cu2 doesn't exist in Qasm. For example in the Qasm paper ([arXiv:1707.03429](https://arxiv.org/pdf/1707.03429.pdf)) both cu1 and cu3 are defined on page 12, however there is no reference to cu2 anywhere in the paper.
Also, when I export the Qasm code to the IBM experience online (for example to plot a circuit), it understands cu1 and cu3 but it doesn't understand cu2.

There are two options: If I add cu2 in the set then our Qasm exports may contain cu2's but these may not be understood by other libraries (like the IBM experience). If I leave it as it is and the user tries to do `to_qasm` in a circuit that contains CU2 it will raise an error that cu2 is not supported by Qasm. I am fine either way.

Reply:
Ok, let's keep the qasm interface compatible with IBM for the time being, we can discuss about that later for the client-server configuration.